provenance_lib/parse.py

```diff
@@ -236,9 +238,49 @@ def __init__(self, archive_fp: str):
         self._number_of_results = 0
 
         with zipfile.ZipFile(archive_fp) as zf:
+            self._get_version_info(zf)
             self._get_root_metadata(zf, archive_fp)
             self._populate_archive(zf)
 
+    def _get_version_info(self, zf: zipfile):
+        """Parse Archive VERSION file"""
+        # This very permissive RE allows 1 to 2-digit integer archive versions,
+        # and any framework version string containing 1 or more non-newline
+        # characters. The final `$` catches VERSION files with too many lines.
+        # _VERSION_MATCHER = r"QIIME 2\narchive: [0-9]{1,2}\nframework: .+$"
```

Comment:
I put together two RE for VERSION file validation. One very permissive (l250), one pretty tightly defined (l257). Descriptions in temporary comments. Any opinions on this?